### PR TITLE
[12.0][MIG] sale_manual_delivery

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
Based on #983 + https://github.com/grindtildeath/sale-workflow/pull/2 at 8th November

Standard migration. Functional tests done from my side and working fine IMHO. Pending todo:

- [x] squash commits
- [x] Check if glue module `sale_stock_picking_blocking_sale_manual_delivery` between this and `sale_stock_picking_blocking`  (proposed in https://github.com/OCA/sale-workflow/pull/935) is really needed to be ported to v12. --> not needed

### CODE REVIEW

**Comment by AaronHForgeFlow** [comment]
cc @grindtildeath

**Review by stevech091 — APPROVED** [review]
functional ok

**Comment by AaronHForgeFlow** [comment]
Added he possibility of changing the route when creating the delivery. This is useful when at the time of creating the order, we don't really know from warehouse and what location we are shipping.

In case no route is provided then the one in the sale order line will be taken.

**Review by rousseldenis — APPROVED** [review]
Code review

**Comment by rousseldenis** [comment]
@AaronHForgeFlow If ready, could you squash ?

**Comment by AaronHForgeFlow** [comment]
@rousseldenis Tried to squash by author and main topics. If the authors of the commits prefer different squashing just let me know @jgrandguillaume @Tonow-c2c @grindtildeath @vrenaville

**Review by leemannd — CHANGES_REQUESTED** [review]
There is a regregression from the first version we did for V10 and V11.

There was the possibility to choose the address form the sale manual delivery wizard.

**Comment by AaronHForgeFlow** [comment]
Hi @leemannd, in the wizard I've seen so far the possibility of changing the carrier and the planned date, but not the delivery address. It is ok to me to add that feature. Could you point me to your repo/the repo where this feature is implemented? Thanks

Also, if I include the delivery_address in the wizard, Would be ok for you to continue with this Pull requests? (same in v10 and v11)

**Comment by leemannd** [comment]
@AaronHForgeFlow Thanks for asking. You have been removing it while helping on the migration of the V11 module -> https://github.com/OCA/sale-workflow/pull/983/commits/1f394ebe123faf580cad03e359093d1f42b31866#diff-ff81e585e998789b0b224c322c8cc742L19

This is present also in the original PR in V10.0 -> https://github.com/OCA/sale-workflow/pull/404/files?file-filters%5B%5D=.xml#diff-ff81e585e998789b0b224c322c8cc742R19

This is for us a blocking point as we have clients using it and needing this possibility.

**Comment by AaronHForgeFlow** [comment]
@leemannd I see. I tried to revert that removal but I missed that part. 

I will add it back here, and in v10/v11

**Comment by leemannd** [comment]
@AaronHForgeFlow Just ping me on them, I will update my reviews right away

**Comment by AaronHForgeFlow** [comment]
@leemannd  Thanks to @mmequignon we have the delivery address back in the wizard :tada:  It would be nice if you check that everything is as expected.

Thanks.

**Comment by AaronHForgeFlow** [comment]
Applied latest fixes in v11

File: sale_manual_delivery/models/stock_move.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Copyright 2017 Denis Leemann, Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, fields, api
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    carrier_id = fields.Many2one("delivery.carrier", string="Delivery Method")
```
Comment (phuctranfxvn) [inline sale_manual_delivery/models/stock_move.py:10]: @AaronHForgeFlow, why do we need to define the carrier at the stock.move level ?
cc @i-vyshnevska

Comment (AaronHForgeFlow) [reply sale_manual_delivery/models/stock_move.py:10]: It is a requirement of the original authors of the module. https://github.com/OCA/sale-workflow/pull/404#discussion_r100574843

Comment (phuctranfxvn) [reply sale_manual_delivery/models/stock_move.py:10]: @AaronHForgeFlow natively Odoo defines the carrier at stock picking level (in delivery module). It confuses us if we have multiple carriers in the same picking (stock.move level), but only one deliver provider defined in picking.

In actual usage, if a picking is delivered by different providers, the picking should be splitted (partial transfer + create back order).

Comment (AaronHForgeFlow) [reply sale_manual_delivery/models/stock_move.py:10]: Splits are not performed anymore with this module. In this case two separate manual deliveries will be created.

Comment (phuctranfxvn) [reply sale_manual_delivery/models/stock_move.py:10]: @i-vyshnevska, do you have any idea?

**Review by leemannd — APPROVED** [review]

**Comment by AaronHForgeFlow** [comment]
@rousseldenis is this fine to be merged?

**Pull request merged**